### PR TITLE
Fixes bugs with Hypothermia & Human Heat Transfer

### LINKED ISSUE #8099: You can be radded through a full radsuit.
(WEB REPORT BY: yred REMOTE: 158.69.55.2:7777)

> Revision 9cb2cff81bce122c4596573433b45605e3d5e52e 

Radsuit and rad hood, stand next to DJ station transmitter, instantly collapse.

### LINKED ISSUE #8101: Test dummy admin spawns (presumably all invulnerable humans?) slowly lose body temperature forever
let me guess, they don't exhaust nutrition so they don't burn calories so they never generate body heat even though they do lose body heat

I think at some point you're gonna have to tweak this burn_calories proc to allow for body heat generation to work independently from nutrient removal so throwing up doesn't cook you and shit

### LINKED ISSUE #8104: Treadmill husks you
I don't care what you say, running on treadmill should never burn you to the point of becoming a husk.

It takes under 5 minutes to kill a man by running on it.

### LINKED ISSUE #8108: Gas mask sprite borked?
(WEB REPORT BY: puglife REMOTE: 158.69.55.2:7777)

> Revision (Should be above if you're viewing this from ingame!)
> 9cb2cff81bce122c4596573433b45605e3d5e52e
> 
> General description of the issue
> Simply put, equipping the grey full face gas mask and moving it out the way replaces my sprite with the default face. Moving it back into place puts eveything back to normal, it's only while the mask is up that it looks funny.

### LINKED ISSUE #8119: Chaplain hoodie sprite switched with something else.
It now has a ugly, fleshy tie on it. And other changes, overall.

How did that happen?

### PR DESCRIPTION
**Pull request by ghost**
- Fixes #8099 
- Fixes #8104 (this one was hilarious, sorry)
- Fixes #8101 
- Fixes #8108 / #8109 
- Fixes #8119
- Fixed the bug where diona heated up infinitely and were immune to the effects of hypothermia (as did monkies, referenced in #7935)
- Made hypothermia & heat transfer more object oriented
- Balanced Basilisks a little better - they now do coldburn damage but their cooling capacity is lowered.

### CODE REVIEW

**Comment by 9600bauds** [comment]
>by adding radiation proof shoes and gloves
I'm not sure if I like this solution
We don't have hardsuit shoes or hardsuit gloves, obviously the suit should just cover those areas

**Comment by ghost** [comment]
@9600bauds *shrug, the sprite is such that that's the intended method.

File: code/modules/clothing/suits/utility.dm [hunk 1]
```diff
@@ -145,3 +145,21 @@
 	slowdown = 1.5
 	armor = list(melee = 0, bullet = 0, laser = 0,energy = 0, bomb = 0, bio = 60, rad = 100)
 	species_fit = list("Vox")
+
+/obj/item/clothing/shoes/radiation
+	name = "radiation shoes"
```
Comment (9600bauds) [inline code/modules/clothing/suits/utility.dm]: I saw that

Comment (ghost) [inline code/modules/clothing/suits/utility.dm]: SHUT UP

File: code/modules/power/treadmill.dm [hunk 2]
```diff
@@ -51,19 +51,17 @@
...
+	if((runner.bodytemperature < T0C + 52) && runner.burn_calories(HUNGER_FACTOR/2))
+		flick("treadmill-running", src)
+		playsound(get_turf(src), 'sound/machines/click.ogg', 50, 1)
+		var/calc = DEFAULT_BUMP_ENERGY * power_efficiency * runner.treadmill_speed
+		if(runner.reagents) //Sanity
+			for(var/datum/reagent/R in runner.reagents.reagent_list)
+				calc *= R.sport
+		if(M_HULK in runner.mutations) calc *= 5
+		count_power += calc
+	else to_chat(runner,"<span class='warning'>You're exhausted! You can't run anymore!</span>")
```
Comment (9600bauds) [inline code/modules/power/treadmill.dm]: CEEEEELT

Comment (ghost) [inline code/modules/power/treadmill.dm]: ?

Comment (9600bauds) [inline code/modules/power/treadmill.dm]: This is hardly just fixing the bug, you're surreptitiously modifying how the treadmill works and that's a case of the d-y-l-a-n-i-t-i-s

Comment (ghost) [inline code/modules/power/treadmill.dm]: It stops them running only if they can't burn enough calories (duh, that makes sense)

_~~b) Are >52C, which takes a fair while to get to now.~~_

**Comment by ghost** [comment]
This is retarded change to treadmill, if you can move you can run. Fix your shitty hypothermia code causing plasma fire level heats.

**Comment by 9600bauds** [comment]
If someone can husk themselves from running that's a glaring flaw in the way heat generation and heat dissipation work. That's not a flaw of the treadmill, that will be the case of every single thing that consumes nutriment faster than normal operation.
You're going to have to unfuck this whole burn_calories() deal at some point.

**Comment by 9600bauds** [comment]
Also yeah if you're too exhausted to run ON the treadmill, how come you can run laps around it just fine

**Comment by ghost** [comment]
@9600bauds It's a flaw in the way that hyperthermia works (In that it doesn't exist). Until such a time that it does, and that will be when I deal with burn_calories() in a more fundamental way, a fix like this, or a fix which caps the temperature gained by the machine is necessary. Ideally, you're right, at 52C you wouldn't be able to run laps at all, but that's for another time.

**Comment by ihadtoregisterforthis** [comment]
Make them collapse, get aches, brute leg damage, not just GTA San Andreas level "You have worked out enough for today". Also what if many people are trying to jog?

**Comment by ghost** [comment]
@ihadtoregisterforthis Good idea actually.

**Comment by ghost** [comment]
Implemented @ihadtoregisterforthis ' solution, you now magically cap out at 99C.

**Comment by ghost** [comment]
Brackets are my mortal enemy.

File: code/modules/power/treadmill.dm [hunk 3]
```diff
@@ -51,19 +51,28 @@
...
+		playsound(get_turf(src), 'sound/machines/click.ogg', 50, 1)
+		var/calc = DEFAULT_BUMP_ENERGY * power_efficiency * runner.treadmill_speed
+		if(runner.reagents) //Sanity
+			for(var/datum/reagent/R in runner.reagents.reagent_list)
+				calc *= R.sport
+		if(M_HULK in runner.mutations) calc *= 5
+		count_power += calc
+		if(runner.bodytemperature > 99)
+			if(prob(5))
+				runner.say("*collapse")
```
Comment (9600bauds) [inline code/modules/power/treadmill.dm]: There's an emote proc just for this, look at how I did it in the brain damage PR yesterday

File: code/modules/power/treadmill.dm [hunk 4]
```diff
@@ -51,19 +51,28 @@
...
+			if(prob(10))
+				to_chat(runner,"<span class='warning'>You really should take a rest!</span>")
+			if(prob(5))
+				to_chat(runner,"<span class='warning'>Your legs really hurt!</span>")
+				runner.apply_damage(2*damage, BRUTE, "l_leg")
+				runner.apply_damage(2*damage, BRUTE, "r_leg")
+			runner.bodytemperature = max(99,cached_temp)
```
Comment (9600bauds) [inline code/modules/power/treadmill.dm]: And we're 100% sure 99 fucking degrees won't still husk you, right?

Comment (ghost) [inline code/modules/power/treadmill.dm]: Yup!

File: code/modules/power/treadmill.dm (hunk 3 above)
Comment (ghost) [inline code/modules/power/treadmill.dm]: Thanks, didn't know that.

File: code/modules/power/treadmill.dm (hunk 4 above)
Comment (ghost) [inline code/modules/power/treadmill.dm]: 100C is where you start to take burn damage as far as I'm aware

Comment (Exxion) [inline code/modules/power/treadmill.dm]: These temperatures are fucking awful but more realistic ones would probably ruin balance to a ridiculous degree

Comment (ghost) [inline code/modules/power/treadmill.dm]: One day I may fix it. Maybe.

**Comment by clusterfack** [comment]
Ok this PR really isn't atomic at all

**Comment by 9600bauds** [comment]
To be quite frank it isn't
for example this entire treadmill/radiation gloves talk is holding back the rest of the fixes
maybe the other bugfixes were fine bundled together because of the nature of the fix but this is a little shaky

**Comment by 9600bauds** [comment]
Of course my shit is blatantly non-atomic but I never get called out on it so hey

**Comment by Exxion** [comment]
You get called out on it sometimes but you just don't care

**Comment by 9600bauds** [comment]
That too

**Comment by FudgePucker** [comment]
Why not just have mag boots yellow gloves and black gloves be rad resistant? And add some black gloves and rad resistant yellow shoes to radsuit lockers. That way you can drag a shard out into space with out 'you are knocked down by air flow' 100 times and also not have to switch around your gloves and shoes to do work with radiation.

**Comment by ghost** [comment]
Bugfixes don't have to be atomic tbh.

Also I realised it wasn't atomic like halfway through

**Comment by ghost** [comment]
@FudgePucker I'll add rad resistance to magboots, but I don't think insulated gloves should be buffed EVEN MORE.

**Comment by FudgePucker** [comment]
The location of the only rad suits on the station is in the bowels of engineering.

It isn't a buff, its a quality of life update after you broke rad suits.

Oh sorry engineers, you have to micro manage gloves because muh balance.

Having rad resistant insulated gloves won't be the biggest buff ever because you are still boned if you don't have rad resistant suit and shoes. You are making it sound like assistants are going to run around with the radio transmitter, in reality engineers have to swap gloves every time they want to change suits.

**Comment by ghost** [comment]
> You broke radsuits

They were broken before in that they covered areas they shouldn't have like the hands and feet.

**Comment by 9600bauds** [comment]
Yes but THEY STILL WORKED ANYWAYS (AND ALWAYS WORKED)
In terms of gameplay, they were perfectly fine.
This is a legacy balance change.
Personally I would just change the radsuit sprite and give it built in gloves and boots.

**Comment by 9600bauds** [comment]
Also, this talk is holding back all the other fixes. Please just make a bunch of separate PRs in the future, unless there is absolutely 0% chance of controversy.

**Comment by ghost** [comment]
To all those complaining about the addition of radiation gloves and radiation boots - My fault.

Celt asked for sprites of the radsuit covering feet and hands, I suggested why not have them as their own seperate thing.

Seems a bit silly, thinking on it

**Comment by Shadowmech88** [comment]
It is. All the other exosuits, namely hardsuits, cover the hands and feet. I don't see why radsuits would be any different.

**Comment by ghost** [comment]
There you go.

**Pull request merged**